### ISSUE TITLE
Proposal: indent NewExpression

### ISSUE THREAD

**Issue by dbellavista** [eslint/eslint#8011]
**What rule do you want to change?**
`indent`: introduce an indentation rule for constructor parameters when using the `new` operator.

**Does this change cause the rule to produce more or fewer warnings?**
More warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option: `NewExpression`, with structure and values similar to the already existing `CallExpression`.

**Please provide some example code that this change will affect:**

```js
/*eslint indent: ["error", 2, {"NewExpression": {"arguments": "first"}}]*/
let o = new MyObject(1,
  2, 3); // <-- Warn, should be aligned with first
```

```js
/*eslint indent: ["error", 2, {"NewExpression": {"arguments": "first"}}]*/
let o = new MyObject(1,
                     2, 3); // <-- Correct
```

**What does the rule currently do for this code?**
The rule should enforce the indentation of the constructor invokation parameter, just like a normal `CallExpression`.

**What will the rule do after it's changed?**

Right now it seems `indent` has no way to specify behaviour when calling the constructor using the `new` operator. This change will affect only those specific cases without affecting other options.

**Comment by not-an-aardvark** [comment]
Is your preferred indentation for `NewExpression` arguments the same as your preferred indentation for `CallExpression` arguments? The [rewrite](https://github.com/eslint/eslint/pull/7618) currently uses the `CallExpression` option to check the indentation of `NewExpression` arguments.

**Comment by dbellavista** [comment]
Yes, I would use the same indentation rules for `NewExpression`.

**Comment by not-an-aardvark** [comment]
Ok, thanks. I'll label this a bug then, since I think it's reasonable to expect `NewExpression` arguments to also be handled by the `CallExpression` option. It's fixed in https://github.com/eslint/eslint/pull/7618, so you can follow that PR for updates.
